### ISSUE TITLE
ESP32: dir(..) on some builtins causes code dump

### ISSUE THREAD

**Issue by robert-hh** [micropython/micropython#6493]
Calling dir(...) on some of the builtins cause a core dump. Tested with no claim on completeness with the following items: 
zip, float, map, range, staticmethod, super, type, StopIteration, Stop AsyncIteration, complex, enumerate, filter, memoryview, reversed, slice, SystemExit, GeneratorExit, all Execption names like TypeError, NameError, ...
ESP8266 fails too.
Pyboard D behaves well, w600 port is fine.

Board Wemos ESP32 Lite, image: esp32-idf3-20200927-unstable-v1.13-77-g0fd0eb00a.bin
MicroPython v1.13-77-g0fd0eb00a on 2020-09-27; ESP32 module with ESP32
Type "help()" for more information.

Backtrace:
Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled.
Core 0 register dump:
PC      : 0x400e8eba  PS      : 0x00060630  A0      : 0x800df42d  A1      : 0x3ffce5d0  
A2      : 0x00000002  A3      : 0x000000b3  A4      : 0x3ffce6b0  A5      : 0x00000000  
A6      : 0x000000b3  A7      : 0x3ffce810  A8      : 0x00000000  A9      : 0x3ffce5d0  
A10     : 0x00000000  A11     : 0x3ffba124  A12     : 0x00000001  A13     : 0x00000000  
A14     : 0x3f40a0b8  A15     : 0x00000001  SAR     : 0x0000001c  EXCCAUSE: 0x0000001c  
EXCVADDR: 0x00000004  LBEG    : 0x4000c46c  LEND    : 0x4000c477  LCOUNT  : 0x00000000  

ELF file SHA256: 0000000000000000000000000000000000000000000000000000000000000000

Backtrace: 0x400e8eba:0x3ffce5d0 0x400df42a:0x3ffce610 0x400df49c:0x3ffce630 0x400eac0d:0x3ffce6b0 0x400e37be:0x3ffce6e0 0x400df709:0x3ffce710 0x400ecf01:0x3ffce730 0x400e3878:0x3ffce7d0 0x400df709:0x3ffce830 0x400df732:0x3ffce850 0x401048bb:0x3ffce870 0x40104ac5:0x3ffce900 0x400f5db8:0x3ffce940 0x4009af06:0x3ffce970

For some reason I could not decode the backtrace

**Comment by jimmo** [comment]
Possibly related to #6306 ? (Haven't tested this myself yet)

**Comment by robert-hh** [comment]
Adding that change indeed  fixes it.

**Comment by mikewadsten** [comment]
> Adding that change indeed fixes it. The question then is: why does it not tip over on the STM32 port.

I'm not 100% sure but I know on other microcontrollers without an MMU (or without using the MMU), things like accessing a null pointer or "writing" to a flash address or bad pointer does not necessarily trigger a hard fault.

**Comment by robert-hh** [comment]
@mikewadsten You may be right, because in that case only read accesses are made.

**Comment by Josverl** [comment]
@robert-hh 
I see issues on the PYBv1.1 as well with the 1.13 build:
MicroPython v1.13 on 2020-09-02; PYBv1.1 with STM32F405RG
```
import io
dir(io.IOBase)
# Expected : 
# ['__class__', '__name__', '__bases__']'
# Actual: 
# MemoryError: memory allocation failed, allocating 1075984344 bytes
```
while it does not crash the CPU, the memory allocation error does not look good either

**Comment by robert-hh** [comment]
Obviously it is a bug, a zero pointer de-reference, which results in different behavior depending on the CPU type.

**Comment by Josverl** [comment]
Agree , just pointed out that it is able to repro on STM32 as well, 
which may make it simpler to add to a test suite without the need to add too much different hardware models>

did not try it on unix though.

**Comment by dpgeorge** [comment]
Thanks for the reports.  Should be fixed by b137d064e9e0bfebd2a59a9b312935031252e742

**Comment by Josverl** [comment]
confirm fixed 
, i tested this on esp32 by running the stubber script that tries to enum just about everything and it runs fine.
